pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -32,10 +32,50 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type ycsbWorkload struct {
+	// workload is the YCSB workload letter e.g. a, b, ..., f.
+	workloadType string
+	// initRows is the number of records to pre-load into the user table.
+	initRows int
+	// waitDuration is the duration the workload should run for.
+	debugRunDuration time.Duration
+	// initSplits is the count of initial splits before resuming work
+	initSplits int
+}
+
+func (ycsb ycsbWorkload) sourceInitCmd(tenantName string, nodes option.NodeListOption) string {
+	cmd := roachtestutil.NewCommand(`./cockroach workload init ycsb`).
+		MaybeFlag(ycsb.initRows > 0, "insert-count", ycsb.initRows).
+		MaybeFlag(ycsb.initSplits > 0, "splits", ycsb.initSplits).
+		Arg("{pgurl%s:%s}", nodes, tenantName)
+	return cmd.String()
+}
+
+func (ycsb ycsbWorkload) sourceRunCmd(tenantName string, nodes option.NodeListOption) string {
+	cmd := roachtestutil.NewCommand(`./cockroach workload run ycsb`).
+		Option("tolerate-errors").
```

Comment:
Curious: why do we always ignore errors?

Reply:
How reliable are these workloads? I wanted to avoid a flaky test and I felt the verifiers we have in place were good enough to catch LDR specific errors

Reply:
Ambiguous errors is the main source of "flakes" I would expect and even then it's debatable whether they should always be ignored in internal testing. Does the cluster run very hot during this test?

Reply:
@stevendanna  thoughts?

Reply:
I think the main thing is we often use these workloads in places where we might also inject node failures (such as in our restart or network outage) tests so we've gotten into the habit of using tolerate errors.

Reply:
It seems like we're already using a `tolerateErrors` variable for the latency verifier itself, presumably for situations like this. We could do the same for the workload.

Reply:
That's true, I can follow up and add an option to the replicateKV struct